Sources/OpenAIKit/Chat/Chat.swift

```diff
@@ -18,21 +18,67 @@ extension Chat {
     public struct Choice {
         public let index: Int
         public let message: Message
-        public let finishReason: String?
+        public let finishReason: FinishReason?
+        
+        public enum FinishReason: String {
+            /// API returned complete model output
+            case stop
+            
+            /// Incomplete model output due to max_tokens parameter or token limit
+            case length
+            
+            /// Omitted content due to a flag from our content filters
+            case contentFilter = "content_filter"
+        }
     }
 }
 
 extension Chat.Choice: Codable {}
 
+extension Chat.Choice.FinishReason: Codable {}
+
 extension Chat {
-    public struct Message {
-        public let role: String
-        public let content: String
-        public init(role: String, content: String) {
-            self.role = role
-            self.content = content
-        }
+    public enum Message {
+        case system(String)
```

Comment:
Let's also add `content` as a named parameter for the associated values for clarity.

Reply:
Done